File: app/src/main/java/com/unify/unifyapp/MainActivity.java
package com.unify.unifyapp;

import android.content.Intent;
import android.os.Bundle;

import androidx.appcompat.app.AppCompatActivity;

import com.unify.avanza.anyline.IUnifyAnyLine;
import com.unify.avanza.app.Unify;
import com.unify.avanza.services.network.ApiClient;
import com.unify.avanza.services.network.IUnifyInterceptor;
import com.unify.avanza.services.network.IUnifyNetworkService;
import com.unify.unifyapp.anyline.ScanMrzActivity;
import com.unify.unifyapp.anyline.ScanUniversalIdActivity;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

import at.nineyards.anyline.core.LicenseException;
import io.anyline.AnylineSDK;

public class MainActivity extends AppCompatActivity implements IUnifyAnyLine, IUnifyNetworkService{

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        Bundle bundle = new Bundle();
//        bundle.putString("url", "https://run.mocky.io/v3/d070b8e2-1c4d-4b65-96d4-82d9f13d383a");
        bundle.putString("url", "http://10.0.2.2:3000/unify_json");
        Unify.getInstance().initialize(this, null, this, this, bundle);

    }

    @Override
    public void onStartPassportScanning() {
        try {
            AnylineSDK.init(getString(R.string.anyline_license_key), this);
        } catch (LicenseException e) {

        }
        startActivity(new Intent(this, ScanMrzActivity.class));
    }

    @Override
    public void onStartUniversalIdScanning() {
        try {
            AnylineSDK.init(getString(R.string.anyline_license_key), this);
        } catch (LicenseException e) {

        }
        startActivity(new Intent(this, ScanUniversalIdActivity.class));
    }

    @Override
    public List<Map<String, IUnifyInterceptor>> addInterceptors() {
        List<Map<String, IUnifyInterceptor>> interceptors = new ArrayList<>();
        Map<String, IUnifyInterceptor> aesInter = new HashMap<>();
        aesInter.put("aes",AESInterceptor.getInstance());
        aesInter.put("checksum",CheckSumInterceptor.getInstance());
        interceptors.add(aesInter);
        return interceptors;
    }
}

File: app/src/main/java/com/unify/unifyapp/MainActivity.kt
package com.unify.unifyapp

import android.os.Bundle
import android.util.Log
import android.view.View
import android.widget.EditText
import androidx.appcompat.app.AppCompatActivity
import com.unify.avanza.EventKeyValue
import com.unify.avanza.UnifyRuntimePermission
import com.unify.avanza.app.Unify
import com.unify.avanza.services.network.IUnifyInterceptor
import com.unify.avanza.services.network.IUnifyNetworkService
import es.dmoral.prefs.Prefs
import org.greenrobot.eventbus.EventBus
import org.greenrobot.eventbus.Subscribe
import org.greenrobot.eventbus.ThreadMode

class MainActivity : AppCompatActivity(), IUnifyNetworkService {

    var url: String = ""
    lateinit var et: EditText
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        EventBus.getDefault().register(this)
        url = Prefs.with(this).read("url","");
        et = findViewById<EditText>(R.id.url)
        et.setText(url)

    }


    override fun addInterceptors(): List<Map<String, IUnifyInterceptor>> {
        val interceptors: MutableList<Map<String, IUnifyInterceptor>> = ArrayList()
        val aesInter: MutableMap<String, IUnifyInterceptor> = HashMap()
        aesInter["aes"] = AESInterceptor
        interceptors.add(aesInter)
        return interceptors
    }

    override fun getValueAgainstBindingKey(): List<String> {
        return listOf("expire", "hideProgress")

    }

    @Subscribe(threadMode = ThreadMode.MAIN)
    fun onMessageEvent(event: EventKeyValue) {
        Log.d("Received in host app", event.bindingKey + " : " + event.value)
    }


    fun initializeSDK(view: View) {
        Prefs.with(view.context).write("url", et.text.toString());

        UnifyRuntimePermission.getPermission(this,
                android.Manifest.permission.WRITE_EXTERNAL_STORAGE,
                object : UnifyRuntimePermission.UnifyPermissionGrantedResponse {
                    override fun onUnifyPermissionGranted() {
                        val dataFromHostapp: MutableMap<String, Any> = HashMap();
                        dataFromHostapp["CPR"] = "00700" // this value provide by host app
                        Unify.getInstance().initializeSDK(this@MainActivity,
                                et.text.toString(),
                                true,
                                dataFromHostapp,
                                null,
                                this@MainActivity)
                        /*Unify.getInstance().initializeSDK(this@MainActivity,
                            "http://mdu-site1.avanzasolutions.com:8010/unifyjsons/Pre_Sales_Onboarding_Master.json",
                            true,
                            dataFromHostapp,
                            null,
                            this@MainActivity)*/
                    }

                })
    }

    fun launchSDK(view: View) {
        UnifyRuntimePermission.getPermission(this,
                android.Manifest.permission.WRITE_EXTERNAL_STORAGE,
                object : UnifyRuntimePermission.UnifyPermissionGrantedResponse {
                    override fun onUnifyPermissionGranted() {
                        Unify.getInstance().launchSDK(this@MainActivity, null);
                    }

                })

    }

}